packages/playwright-core/src/client/browserContext.ts

```diff
@@ -237,7 +237,7 @@ export class BrowserContext extends ChannelOwner<channels.BrowserContextChannel>
     await this._channel.setHTTPCredentials({ httpCredentials: httpCredentials || undefined });
   }
 
-  async addInitScript(script: Function | string | { path?: string, content?: string }, arg?: any): Promise<void> {
+  async addInitScript<Arg>(script: structs.PageFunction<Arg, void> | { path?: string, content?: string }, arg?: Arg): Promise<void> {
```

Comment:
The change should instead go [here](https://github.com/microsoft/playwright/blob/main/utils/generate_types/overrides.d.ts#L30) and [here](https://github.com/microsoft/playwright/blob/main/utils/generate_types/overrides.d.ts#L97).

Ideally, a test snippet [like this](https://github.com/microsoft/playwright/blob/main/utils/generate_types/test/test.ts#L574-L576) would be great to ensure types work as expected.

Make sure to run `npm run watch` and `npm run lint` to check that everything works.

Reply:
I've made the changes in the overrides file and added a test

I had to add line breaks in existing tests that use `addInitScript` to ensure their script function returned void